feat(billing): Update spend notifications documentation (#10076)
getsentry/sentry-docs: docs/product/alerts/notifications/notification-settings.mdx

- **Issue Workflow:** Changes in issue assignment, resolution status, and comments.
- **Deploys:** Release, environment, and commit overviews.
- **Nudges:** Notifications that require review or approval.

> - **Spend:** Enable or disable receiving your organization's spend notifications.

Reviewer: ```suggestion
- **Spend:** Notifications alerting you about your organization's spend and billing quota. 
```
